### PR TITLE
SLEEP-1499 Support embedding of Superset dashboard in Iaso

### PR DESCRIPTION
**Pull request by bramj**
This PR contains a v1 for embedding Superset dashboards in Iaso, using the `Pages` model.

Specifically:
- Add a new pages template that supports embedding a new type of page "SUPERSET"
- Add an API endpoint POST `/api/superset/token` that creates a Superset "guest token" for a specific dashboard

Related JIRA tickets : https://bluesquare.atlassian.net/browse/SLEEP-1499

**Note that this is a v1. In order to make this properly usable for regular users, I think we should have a place where you can configure the connection to such a dashboard for a given account.**

image

## How to test

1. Create a Superset dashboard, make it embeddable, this will give you a `dashboard_id`.
2. Create an embedded page of type "Superset". Set the `dashboard_id` of the dashboard you want to embed.
3. Visit the page, you should now see your dashboard.
4. In the admin page you can configure the UI config options for your dashboard.
example config:
```json
{
  "hideTab": true,
  "hideTitle": true,
  "hideChartControls": true,
  "filters": {
    "visible": false,
    "expanded": false
  },
}
```
(On Trypelim, I have only the `hideTab` and `hideTitle` set. Filters are useful and chart controls allow things like exporting CSV data)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -82,7 +82,17 @@ services:
       WFP_EMAIL_RECIPIENTS_NEW_ACCOUNT:
       DISABLE_PASSWORD_LOGINS:
       MAINTENANCE_MODE:
-      SERVER_URL: # Limit logging in dev to not overflow terminal
+      SERVER_URL:
+      SUPERSET_DB_NAME:
+      SUPERSET_DB_HOSTNAME:
+      SUPERSET_DB_PASSWORD:
```
Comment (madewulf) [inline docker-compose.yml]: Why is that here?

File: hat/templates/iaso/pages/superset.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<!doctype html>
+<html>
+    <head>
+        <title>{{title}}</title>
+        {{ analytics_script | safe }}
+        <script src="https://unpkg.com/@superset-ui/embedded-sdk"></script>
```
Comment (madewulf) [inline hat/templates/iaso/pages/superset.html]: Ideally, we should have this copied in our code base and not depending on an external source

File: hat/templates/iaso/pages/superset.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                return json_resp.token;
+            }
+
+            const containerRef = document.getElementById(
+                'my-superset-container',
+            );
+
+            supersetEmbeddedSdk.embedDashboard({
+                id: '{{dashboard_id}}',
+                supersetDomain: 'https://superset.trypelim.org',
```
Comment (madewulf) [inline hat/templates/iaso/pages/superset.html]: Domain name seems hardcoded?

File: hat/templates/iaso/pages/superset.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
+<!doctype html>
+<html>
+    <head>
+        <title>{{title}}</title>
+        {{ analytics_script | safe }}
+        <script src="https://unpkg.com/@superset-ui/embedded-sdk"></script>
+    </head>
+
+    <body>
+        <div id="my-superset-container"></div>
```
Comment (madewulf) [inline hat/templates/iaso/pages/superset.html]: Not very important, but why "my" ?

**Comment by bramj** [comment]
@beygorghor What do you think of this commit? https://github.com/BLSQ/iaso/pull/1698/commits/0ef8748d57d1b5e7cfde10df7e696f67ff96fc92

@madewulf Okay for re-review!

File: hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js [hunk 5]
```diff
@@ -233,13 +233,27 @@ const CreateEditDialog = ({ isOpen, onClose, selectedPage }) => {
...
+                                            <>
+                                                <Field
+                                                    label={formatMessage(
+                                                        MESSAGES.superset_dashboard_id,
+                                                    )}
+                                                    name="superset_dashboard_id"
+                                                    component={TextInput}
+                                                    className={classes.input}
+                                                />
+                                            </>
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js]: this fragment is useless

**Review by beygorghor — APPROVED** [review]

File: hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js (hunk 5 above)
Comment (bramj) [reply hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js]: Ah yes indeed, I had 2 inputs at some point, forgot to clean this up

**Review by madewulf — APPROVED** [review]
